101-application-gateway-rewrite/azuredeploy.json

```diff
@@ -0,0 +1,312 @@
+      "apiVersion": "[parameters('KeyVaultVersion')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.ManagedIdentity/userAssignedIdentities/', variables('idName'))]"
+      ],
+      "properties": {
+        "sku": {
+          "name": "standard",
+          "family": "A"
+        },
+        "tenantId": "[reference(variables('identityID'), parameters('MSIVersion')).tenantId]",
+        "accessPolicies": [
+          {
+            "tenantId": "[reference(variables('identityID'), parameters('MSIVersion')).tenantId]",
+            "objectId": "[reference(variables('identityID'), parameters('MSIVersion')).principalId]",
+            "permissions": {
+              "secrets": [
+                "get"
+              ]
+            }
+          }
+        ],
+        "enableSoftDelete": true
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/secrets",
+      "dependsOn": [
+        "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+      ],
+      "apiVersion": "[parameters('KeyVaultVersion')]",
+      "copy": {
+        "name": "secretcopy",
+        "count": "[length(parameters('Secrets'))]"
+      },
+      "name": "[concat(variables('keyVaultName'), '/', parameters('Secrets')[copyIndex()].name)]",
+      "properties": {
+        "value": "[parameters('Secrets')[copyIndex()].value]",
+        "recoveryLevel": "Purgeable"
+      }
+    },
+    {
+      "apiVersion": "[variables('apiVersion')]",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "name": "[variables('publicIPAddressName')]",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "Standard"
+      },
+      "properties": {
+        "publicIPAllocationMethod": "Static"
+      }
+    },
+    {
+      "apiVersion": "[variables('apiVersion')]",
+      "type": "Microsoft.Network/virtualNetworks",
+      "name": "[variables('virtualNetworkName')]",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[parameters('addressPrefix')]"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "[variables('subnetName')]",
+            "properties": {
+              "addressPrefix": "[parameters('subnetPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "[variables('apiVersion')]",
+      "name": "[variables('applicationGatewayName')]",
+      "type": "Microsoft.Network/applicationGateways",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+        "[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]",
+        "[concat('Microsoft.KeyVault/vaults/', variables('keyVaultName'))]"
+      ],
+      "identity": {
+        "type": "UserAssigned",
```

Comment:
why not system assigned, if we're just creating one anyway?

Reply:
Just preferred to keep it userAssigned since we are defining a fixed static value everytime and is not getting autogenerated by system.

Reply:
That's not a good reason for the use case... user assigned means the user has to provide principal, if that's not required, then let's use system assigned.